### DISCUSSION TITLE
Fundamentos do ASP.NET 6 > CRUD e Entity Framework > ViewModels

### DISCUSSION THREAD

**Discussion by DrenokBR** [balta-io/forum#463] [Tira Dúvidas] [answered]
Fala André !

Se não quisermos usar a abordagem "MVVM" (com as ViewModels) para resolver o problema de entradas indesejadas através do padrão JSON podemos usar o "Data Annotation" [JsonIgnore]. Parece que sim, né ? E se for realmente viável, misturar isso com o paradígma "Fluent Mapping" seria um "Code Smell" ou mesmo usando o Fluent Mapping podemos usar um atributo ou outro ?!

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @DrenokBR, como vai?
Sim é possível utilizar ViewModels e ter as validações nela, tanto com Data Annotations quanto com Fluent Validator (Ou Flunt).
O Fluent Mapping é para mapeamento de Entidades ou Modelos, eles não afetam (Ou não deveriam afetar) as ViewModels.
